### PR TITLE
PAE-628 Add MIT HZ integration

### PR DESCRIPTION
**Pull request by luismorenolopera**
Description
---------------
This PR adds a new integration with MIT Horizon.

**Ticket:** [PAE-628](https://pearsonadvance.atlassian.net/browse/PAE-628)

### CODE REVIEW

File: openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        }
+        try:
+            response = requests.post(
+                url=url,
+                json=data,
+            )
+        except Exception as error:
+            msg_error = 'Failed to authenticate at MIT HZ API. Reason: %s'
+            log_details['response'] = {'error': msg_error + str(error)}
+            LOG.error(msg_error + str(error))
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py:58]: Can we concatenate the error message in a single statement? @luismorenolopera 
```suggestion
            msg_error = 'Failed to authenticate at MIT HZ API. Reason: %s' % str(error)
            log_details['response'] = {'error': msg_error}

            LOG.error(msg_error)
```

File: openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        log_details = {
+            'url': url,
+            'user': user_id,
+        }
+
+        try:
+            response = requests.get(url, headers=headers)
+        except Exception as error:
+            LOG.error('Failed to get user at MIT HZ API.')
+            log_details['response'] = {'error': 'Failed to get user at MIT HZ API. Reason: %s' + str(error)}
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py:102]:
```suggestion
            log_details['response'] = {'error': 'Failed to get user at MIT HZ API. Reason: %s' % str(error)}
```

File: openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        else:
+            if not response.ok:
+                log_details['response'] = response.text()
+                EnrollmentRequestLog.objects.create(
+                    request_type=str(self),
+                    details=log_details,
+                )
+                LOG.error('failed when trying to authenticate with MIT HORIZON')
+                return 'missing token'
+            return response.json()['access_token']
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py:72]: Can we add a blank line above the return statement? Also separate the assignment and action statements in all occurrences? @luismorenolopera.

File: openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                details=log_details,
+            )
+            return True
+
+    def _get_enrollment_data(self, data, course_settings):
+        """
+        Returns the data required to refresh a user in the MIT HORIZON API.
+        """
+        enrollment_data = {}
+        user_id = self._get_user_id(data.get('user_email'))
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py:131]: Does '_get_user_id' email param nullable? @luismorenolopera.

File: openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+    def _check_user(self, user_id):
+        """
+        Check if user exists in MIT HORIZON.
+        """
+        headers = self._get_enrollment_headers()
+        url = '{root_url}{get_user_path}{user_id}'.format(
+            root_url=settings.MIT_HZ_API_URL,
+            get_user_path=settings.MIT_HZ_GET_USER_PATH,
+            user_id=user_id
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py:91]: Please can we add a trailing coma? @luismorenolopera.

**Review by ivanvgh — CHANGES_REQUESTED** [review]

File: openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py (hunk 4 above)
Comment (luismorenolopera) [reply openedx_external_enrollments/external_enrollments/mit_hz_external_enrollment.py:131]: I don't think this case is possible @ivanvgh

**Review by diegomillan — DISMISSED** [review]

**Review by ivanvgh — DISMISSED** [review]

**Review by ivanvgh — APPROVED** [review]

**Pull request merged**